### PR TITLE
Upgrade spacy version

### PR DESCRIPTION
**Pull request by SharonHart**
Upgrade Spacy to 3.0.5

### CODE REVIEW

File: presidio-analyzer/tests/test_analyzer_engine.py [hunk 1]
```diff
@@ -273,7 +273,7 @@ def test_when_entities_is_none_all_recognizers_loaded_then_return_all_fields(
         registry=RecognizerRegistry(), nlp_engine=nlp_engine
     )
     threshold = 0
-    text = "My name is David and I live in Seattle." "Domain: microsoft.com "
+    text = "My name is Sharon and I live in Seattle." "Domain: microsoft.com "
```
Comment (shiranr) [inline presidio-analyzer/tests/test_analyzer_engine.py:276]: 😆

File: presidio-analyzer/Pipfile [hunk 2]
```diff
@@ -4,7 +4,7 @@ verify_ssl = true
 name = "pypi"
 
 [packages]
-spacy = "==2.3.4"
+spacy = "==3.0.5"
```
Comment (shiranr) [inline presidio-analyzer/Pipfile:7]: setup.py also needs to be updated.

File: presidio-analyzer/tests/test_analyzer_engine.py (hunk 1 above)
Comment (SharonHart) [reply presidio-analyzer/tests/test_analyzer_engine.py:276]: For some reason David wasn't recognized. *I plan to stay on the Israeli site

Comment (shiranr) [reply presidio-analyzer/tests/test_analyzer_engine.py:276]: Well, David is a very basic name. that is very weird. I would add @omri374 to this conversation.

File: presidio-analyzer/Pipfile (hunk 2 above)
Comment (SharonHart) [reply presidio-analyzer/Pipfile:7]: Done

File: presidio-analyzer/tests/test_analyzer_engine.py (hunk 1 above)
Comment (omri374) [reply presidio-analyzer/tests/test_analyzer_engine.py:276]: :) https://github.com/explosion/spaCy/issues/7493#issuecomment-803459979

File: presidio-analyzer/setup.py [hunk 3]
```diff
@@ -31,7 +31,7 @@
     trusted_host=["pypi.org"],
     tests_require=["pytest", "flake8==3.7.9"],
     install_requires=[
-        "spacy==2.3.4",
+        "spacy==3.0.5",
```
Comment (omri374) [inline presidio-analyzer/setup.py:34]: how was the dataclasses issue fixed?

Comment (SharonHart) [reply presidio-analyzer/setup.py:34]: Locked with python 3.6

File: presidio-analyzer/tests/test_analyzer_engine.py (hunk 1 above)
Comment (shiranr) [reply presidio-analyzer/tests/test_analyzer_engine.py:276]: @omri374 and we still want to upgrade?

**Review by navalev — APPROVED** [review]
LGTM

**Review by shiranr — APPROVED** [review]

**Pull request merged**